packages/hoppscotch-common/src/components/http/Request.vue

```diff
@@ -593,7 +597,12 @@ defineActionHandler("request.method.prev", cycleUpMethod)
 defineActionHandler("request.save", saveRequest)
 defineActionHandler(
   "request.save-as",
-  () => (showSaveRequestModal.value = true)
+  (req: { request: HoppRESTRequest | HoppGQLRequest }) => {
+    showSaveRequestModal.value = true
+    if (req) {
+      request.value = req.request
+    }
+  }
```

Comment:
Since, `request` can be a `HoppGQLRequest`, make sure that situation is validated.